shell/platform/linux/fl_accessible_node.cc

```diff
@@ -0,0 +1,385 @@
+    {ATK_STATE_FOCUSED, kFlutterSemanticsFlagIsFocused, FALSE},
+    {ATK_STATE_CHECKED, kFlutterSemanticsFlagIsChecked, FALSE},
+    {ATK_STATE_SELECTED, kFlutterSemanticsFlagIsSelected, FALSE},
+    {ATK_STATE_ENABLED, kFlutterSemanticsFlagIsEnabled, FALSE},
+    {ATK_STATE_READ_ONLY, kFlutterSemanticsFlagIsReadOnly, FALSE},
+    {ATK_STATE_INVALID, static_cast<FlutterSemanticsFlag>(0), FALSE},
+};
+
+// Maps Flutter semantics actions to ATK actions.
+typedef struct {
+  FlutterSemanticsAction action;
+  const gchar* name;
+} ActionData;
+static ActionData action_mapping[] = {
+    {kFlutterSemanticsActionTap, "Tap"},
+    {kFlutterSemanticsActionLongPress, "LongPress"},
+    {kFlutterSemanticsActionScrollLeft, "ScrollLeft"},
+    {kFlutterSemanticsActionScrollRight, "ScrollRight"},
+    {kFlutterSemanticsActionScrollUp, "ScrollUp"},
+    {kFlutterSemanticsActionScrollDown, "ScrollDown"},
+    {kFlutterSemanticsActionIncrease, "Increase"},
+    {kFlutterSemanticsActionDecrease, "Decrease"},
+    {kFlutterSemanticsActionShowOnScreen, "ShowOnScreen"},
+    {kFlutterSemanticsActionMoveCursorForwardByCharacter,
+     "MoveCursorForwardByCharacter"},
+    {kFlutterSemanticsActionMoveCursorBackwardByCharacter,
+     "MoveCursorBackwardByCharacter"},
+    {kFlutterSemanticsActionSetSelection, "SetSelection"},
+    {kFlutterSemanticsActionCopy, "Copy"},
+    {kFlutterSemanticsActionCut, "Cut"},
+    {kFlutterSemanticsActionPaste, "Paste"},
+    {kFlutterSemanticsActionDidGainAccessibilityFocus,
+     "DidGainAccessibilityFocus"},
+    {kFlutterSemanticsActionDidLoseAccessibilityFocus,
+     "DidLoseAccessibilityFocus"},
+    {kFlutterSemanticsActionCustomAction, "CustomAction"},
+    {kFlutterSemanticsActionDismiss, "Dismiss"},
+    {kFlutterSemanticsActionMoveCursorForwardByWord, "MoveCursorForwardByWord"},
+    {kFlutterSemanticsActionMoveCursorBackwardByWord,
+     "MoveCursorBackwardByWord"},
+    {static_cast<FlutterSemanticsAction>(0), nullptr}};
+
+struct _FlAccessibleNode {
+  AtkObject parent_instance;
+
+  FlEngine* engine;
+  AtkObject* parent;
+  int32_t id;
+  gchar* name;
+  gint x, y, width, height;
+  GPtrArray* actions;
+  gsize actions_length;
+  GPtrArray* children;
+  FlutterSemanticsFlag flags;
+};
+
+static void fl_accessible_node_component_interface_init(
+    AtkComponentIface* iface);
+static void fl_accessible_node_action_interface_init(AtkActionIface* iface);
+static void fl_accessible_node_text_interface_init(AtkTextIface* iface);
+
+G_DEFINE_TYPE_WITH_CODE(
+    FlAccessibleNode,
+    fl_accessible_node,
+    ATK_TYPE_OBJECT,
+    G_IMPLEMENT_INTERFACE(ATK_TYPE_COMPONENT,
+                          fl_accessible_node_component_interface_init)
+        G_IMPLEMENT_INTERFACE(ATK_TYPE_ACTION,
+                              fl_accessible_node_action_interface_init)
+            G_IMPLEMENT_INTERFACE(ATK_TYPE_TEXT,
+                                  fl_accessible_node_text_interface_init))
```

Comment:
Out of curiosity, what does this interface macro expand to that it doesn't need any separators between the invocations?

Reply:
It expands to a `g_type_add_interface_static()` call - the `G_DEFINE_TYPE` macros expands to make the `fl_accessible_node_get_type()` function.